tests/structured-data/class-archive-post-test.php

```diff
@@ -233,6 +233,143 @@ public function test_term_archive() {
 		self::assertEquals( $cat_link, $structured_data['url'] );
 	}
 
+	/**
+	 * Check metadata for custom post type archive.
+	 *
+	 * @covers \Parsely::construct_parsely_metadata
+	 * @uses \Parsely::__construct
+	 * @uses \Parsely::get_author_name
+	 * @uses \Parsely::get_author_names
+	 * @uses \Parsely::get_bottom_level_term
+	 * @uses \Parsely::get_category_name
+	 * @uses \Parsely::get_clean_parsely_page_value
+	 * @uses \Parsely::get_coauthor_names
+	 * @uses \Parsely::get_current_url
+	 * @uses \Parsely::get_first_image
+	 * @uses \Parsely::get_options
+	 * @uses \Parsely::get_tags
+	 * @uses \Parsely::post_has_trackable_status
+	 * @uses \Parsely::update_metadata_endpoint
+	 * @group metadata
+	 */
+	public function test_archive_custom_post_type() {
+		// Set permalinks, as Parsely currently strips ?page_id=... from the URL property.
+		// See https://github.com/Parsely/wp-parsely/issues/151
+		$this->set_permalink_structure( '/%postname%/' );
+
+		// Setup Parsley object.
+		$parsely         = new \Parsely();
+		$parsely_options = get_option( \Parsely::OPTIONS_KEY );
+
+		// Register Post Type with specific archive URL.
+		register_post_type(
+			'custom_post_type',
+			array(
+				'public'      => true,
+				'has_archive' => 'cpt-tax-archive',
```

Comment:
If you've got a CPT, why give it an archive of `cpt-tax-archive`? The `-tax` bit seems irrelevant since there's no Custom Taxonomy aspect of this test.